src/argprocessing.cpp

```diff
@@ -1423,22 +1416,39 @@ process_args(Context& ctx)
     compiler_args.push_back("-dc");
   }
 
-  if (state.found_xarch_arch && !args_info.arch_args.empty()) {
-    if (args_info.arch_args.size() > 1) {
+  if (!state.xarch_args.empty()) {
+    if (args_info.arch_args.size() != state.xarch_args.size()) {
```

Comment:
If I understand the relationship between `-arch` and `-Xarch` correctly, `-Xarch` options are only relevant for the matching `-arch`, which is why we can transform

```
-arch A -Xarch_A a -arch B -Xarch_B b
```

to two preprocessor calls with

```
-arch A -Xarch_A a
-arch B -Xarch_B b
```

Does that sound correct?

Since the code now already handles the logic of matching `-Xarch` options with `-arch`, perhaps it would be just as easy to support zero or multiple `-Xarch` options? That is, could we transform

```
-arch A -Xarch_A a1 -Xarch_A a2 -arch B -Xarch_B b -arch C
```

into

```
-arch A -Xarch_A a1 -Xarch_A a2
-arch B -Xarch_B b
-arch C
```

?

Reply:
Honestly I'm not sure if the third example should work. I am guessing here a bit the relationship between `-arch A` and `-Xarch_A`. Will try to find some documentation so I feel more confident in my change.

Reply:
All three options above are supported now.